python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -947,7 +953,16 @@ def forward(self, hidden_states: torch.Tensor, router_logits: torch.Tensor):
             ),
         )
 
-        if self.reduce_results and (self.tp_size > 1 or self.ep_size > 1):
+        if (
+            _is_flashinfer_available
+            and global_server_args_dict["enable_flashinfer_allreduce"]
+            and hidden_states.shape[0] <= 128
+            ):
+            from sglang.srt.layers.flashinfer_comm import (
+                flashinfer_allreduce,
+            )
+            final_hidden_states = flashinfer_allreduce(final_hidden_states)
```

Comment:
Shouldn't this check and replacement be under `self.reduce_results and (self.tp_size > 1 or self.ep_size > 1)`?